### ISSUE TITLE
How to handle UDT socket.connect broken connection

### ISSUE THREAD

**Issue by judef** [dump247/udt-net#27]
I have a web application which uses UDT.Net to upload files and Another console application that runs on server to receive the files. The below code is part of the handler which asynchronously(threading) uploads file as & when a user selects them, the issue is when the server console application is down, socket.connect(as shown in below code) from web app throws an exception and w3wp.exe(IIS) crashes. 
		How can I handle the ``Connection setup failure: connection timeout,
UDT Error Code: 1001 `` error so that IIS doesn`t crash. Any help would be greatly appreciated? Thank once again for  UDT.net

```vbnet
Public Sub UDPUpload(ByVal f As UploadedFile)
        Try
             
            Dim remoteHostEntry As IPHostEntry = Dns.GetHostEntry(UDPSERVERNAME)

            'receiver to connect and save the files on webserver
            Using client As New Udt.Socket(AddressFamily.InterNetwork, SocketType.Stream)
                client.Connect(clsDOC.ReturnIP4Address(remoteHostEntry), Integer.Parse(UDPPORT))
               'code that sends the file to consolde app
                '.............................................

                ClientRec(client, strRec, 0, 4)
                client.Close()
            End Using
      Catch ex As Exception
           
        End Try
    End Sub
```

**Comment by dump247** [comment]
I am not sure what help you need here. The udt.net library will throw a `Udt.SocketException` if any error occurs. You are clearly trapping and swallowing all exceptions, so IIS clearly shouldn't be crashing because of an exception in the udt library. Are you saying that IIS crashes immediately after that error occurs, even if you handle the error? Does it happen every time you run it (i.e. is it easily repeatable)?

**Comment by judef** [comment]
Yup, even if we handle the error, IIS crashes imediately. It happens every time when the above error is thrown. Below is the IIS log. Thank you for all the help.

![2016-02-18_1901](https://cloud.githubusercontent.com/assets/63353/13147418/61d46188-d672-11e5-9251-88c7bcb51804.png)

**Comment by dump247** [comment]
It appears to be an issue in the underlying native udt library. This .NET library is a fairly thin wrapper around the native library, which does all the real work.

See https://sourceforge.net/p/udt/discussion/393037/thread/e15eb3b4/

The udt project appears to be dead. I say "appears" because there have been no updates for 3+ years and ~~the author hasn't responded to my request to update the link to this project~~ (nvm, the link got updated).

**Comment by judef** [comment]
Oh ok, thank you for the info. Please let us know if there is a workaround, will also try our best to look into native C++ library. Once again thank you for UDT.Net
